xarray/core/coordinates.py

```diff
@@ -561,6 +561,31 @@ def merge(self, other: Mapping[Any, Any] | None) -> Dataset:
             variables=coords, coord_names=coord_names, indexes=indexes
         )
 
+    def __or__(self, other: Self) -> Self:
+        """Merge two sets of coordinates to create a new Coordinates object
```

Comment:
Do we need such docstrings for an operator?

Reply:
not really, but it serves as a comment in this case. No need to copy everything, though, so we can remove everything but the first line and add a `See Also` section to `Coordinates.merge`

Reply:
It looks like we're generating documentation pages for `__setitem__` and __getitem__`. Not sure if this actually makes sense, but following that I added a page for `__or__` so we actually do need the full docstring.